### ISSUE TITLE
Transfer workspace settings overrides when overwriting existing GeoServer workspace

### ISSUE THREAD

**Issue by kannes** [GeoCat/qgis-bridge-plugin#177]
**Describe the bug**
Similar to #168, #169 and #176 there are also workspace-specific settings overrides possible. E.g. for contact information or proxy setup.

![image](https://github.com/GeoCat/qgis-bridge-plugin/assets/7661092/14f1ca19-5e04-4f8d-9e8b-9ceef0205798)

Currently, if a workspace has overridden settings, they are dropped when re-creating the workspace.

The plugin should try to preserve them.

**Additional context**
Sorry for dropping these step by step, I am learning GeoServer's funkiness as I go :} Please do in no way feel obligated to respond quickly!

I have *not* yet got this working because the GeoServer API will return some values even if the setting is not actually enabled: https://gis.stackexchange.com/questions/479679/querying-geoserver-for-workspace-specific-local-settings
I'll keep an eye on the issue and implement it once possible.

**Comment by GeoSander** [comment]
Indeed, I am aware that there may be workspace-specific settings (and that you can obtain them via the `{workspace}/settings.json` endpoint). Already had the feeling that you were about to bring that up 😅 

I don't think that it matters if settings are enabled in the admin UI or not. If you want to support this, I think you should:

1. Get the existing workspace details (i.e. ACL, settings, etc.);
2. Recreate the workspace (that will create a default settings object)
3. Do a `PUT` on the `{workspace}/settings.json` endpoint with the retrieved `settings` object from step 1. That should update the settings accordingly.

I believe that you actually always need to check the `Enabled` checkbox in the GeoServer admin UI if you want to see or edit the settings. It's not that you can really switch them off.

**Comment by kannes** [comment]
I had tried the same but unfortunately that will lead to the default settings overriding the global settings, even if they are mostly empty.

- I set the organisation to "foo" in the global Contact Information settings.
- A GetCapabilities request to any workspace without settings override now includes `<ContactOrganization>foo</ContactOrganization>`
- If I request settings.json from such a workspace, I get:
```
 workspace_settings={
  'settings': {
    'contact': {'id': 'contact'},
    'charset': 'UTF-8',
    'numDecimals': 4,
    'verbose': False,
    'verboseExceptions': False,
    'localWorkspaceIncludesPrefix': False,
    'showCreatedTimeColumnsInAdminList': False, 
    'showModifiedTimeColumnsInAdminList': False
  }
}
```
- If I apply that to a new workspace, the GUI will show the checkbox enabled, with all blank fields for the Contact Information. And if I request a GetCapabilities document of that workspace, I get `
<ContactOrganization/>` instead of the global `<ContactOrganization>foo</ContactOrganization>`.

A workaround could be to set an empty `'settings': {}` if the default dict was received but that would be a bug if someone happened to configure their workspace override exactly like the default :B

I will try to find out more, maybe something else must be set somewhere.
